app/js/components/LabResultsList.jsx

```diff
@@ -327,7 +327,12 @@ export class LabResultsList extends PureComponent {
     const getPatientLabResults = () => {
       // build a list of all obs from the encounters that are of type LabSet or Test
       return encounters.reduce((acc, encounter) => {
-        const obs = encounter.obs ? encounter.obs.filter(o => isLabSet(o) || isTest(o)) : [];
+        let obs = encounter.obs ? encounter.obs : [];
+        // flatten obs groups into a single list of obs
+        while (obs.some(o => o.groupMembers)) {
```

Comment:
Isn't the some() function returning immediately when it encounters the first obsGroup? So, if there are multiple obsGroups in the same encounter, it only processes the first obsGroup?

Reply:
Yes, but some just returns a boolean, so this is basically saying "while there's still one or more nested obs groups, run the the flatMap again"... instead of how I arbitrarily ran the flatMap 3 times before... each flat map should effectively flatten one level of obs groups.

Reply:
Got it! Thanks @mogoodrich !